tokio/src/time/driver/handle.rs

```diff
@@ -24,42 +22,45 @@ thread_local! {
 }
 
 #[derive(Debug)]
-///Unsets default timer handler on drop.
+/// Guard that resets the default timer on drop.
 pub(crate) struct DefaultGuard<'a> {
-    prev: Option<HandlePriv>,
     _lifetime: PhantomData<&'a u8>,
 }
 
 impl Drop for DefaultGuard<'_> {
     fn drop(&mut self) {
         CURRENT_TIMER.with(|current| {
             let mut current = current.borrow_mut();
-            *current = self.prev.take();
+            *current = None;
         })
     }
 }
 
-///Sets handle to default timer, returning guard that unsets it on drop.
+/// Sets handle to default timer, returning guard that unsets it on drop.
 ///
 /// # Panics
 ///
 /// This function panics if there already is a default timer set.
 pub(crate) fn set_default(handle: &Handle) -> DefaultGuard<'_> {
     CURRENT_TIMER.with(|current| {
         let mut current = current.borrow_mut();
-        let prev = current.take();
```

Comment:
I think this should be reverted and punted to a separate discussion. I was working on allowing re-entry (and I think some tests need it). Re-entry is needed since there are no more APIs that take handles, so it is the only way to use a custom timer.